packages/ckeditor5-find-and-replace/src/utils.js

```diff
@@ -138,7 +133,15 @@ export function findByTextCallback( searchTerm, options ) {
 	if ( options.wholeWords ) {
 		const nonLetterGroup = '[^a-zA-Z\u00C0-\u024F\u1E00-\u1EFF]';
 
-		regExpQuery = `(^|${ nonLetterGroup }|_)` + regExpQuery + `(?:_|${ nonLetterGroup }|$)`;
+		if ( !new RegExp( '^' + nonLetterGroup ).test( searchTerm ) )
+		{
+			regExpQuery = `(?<=^|${ nonLetterGroup }|_)${ regExpQuery }`;
```

Comment:
This option is unfortunately not supported by Safari so we need to find a better way.

Reply:
Reverted to the previous mechanism.